src/tools/opt-dist/src/utils/io.rs

```diff
@@ -46,3 +53,28 @@ pub fn unpack_archive(path: &Utf8Path, dest_dir: &Utf8Path) -> anyhow::Result<()
     archive.unpack(dest_dir.as_std_path())?;
     Ok(())
 }
+
+/// Finds a single file in the specified `directory` with the given `prefix` and `suffix`.
+pub fn find_file_in_dir(
+    directory: &Utf8Path,
+    prefix: &str,
+    suffix: &str,
+) -> anyhow::Result<Utf8PathBuf> {
+    let mut files = glob::glob(&format!("{directory}/{prefix}*{suffix}"))?
+        .into_iter()
+        .collect::<Result<Vec<_>, _>>()?;
+    match files.pop() {
+        Some(file) => {
+            if !files.is_empty() {
```

Comment:
Not hard feelings: Would be easier to read this with less nested blocks by `Some(file) if !files.is_empty()`

Reply:
Or check for the length !=1 errors before the match, removing the need to push back the thing you popped to restore the original list.